docs/orchestrating-elastic-stack-applications/beat.asciidoc

```diff
@@ -484,16 +493,16 @@ Deploys Packetbeat as a DaemonSet that monitors DNS on port `53` and HTTP(S) tra
 [id="{p}-beat-beat-pods-are-crashing-when-kibanaref-is-specified"]
 === Beat Pods are crashing when kibanaRef is specified
 
-When `kibanaRef` is specified, Beat tries to connect to the Kibana instance. If it's unable to do so, the Beat process exits and the Pod restarts. This may happen when Kibana is not yet up or when Beat user is not yet created in Elasticsearch. The Pod may restart a few times when it is first deployed. Afterwards, the Beat should run successfully.
+When `kibanaRef` is specified, Beat tries to connect to the Kibana instance. If it's unable to do so, the Beat process exits and the Pod restarts. This may happen when Kibana is not yet up or when a Beat user is not yet created in Elasticsearch. The Pod may restart a few times when it is first deployed. Afterwards, the Beat should run successfully.
 
 
 [id="{p}-beat-configuration-containing-key-null-is-malformed"]
 === Configuration containing key: null is malformed
 
-When `kubectl` is used to modify a resource, it calculates a diff between the user applied and the already existing config. This diff has a special semantics link:https://tools.ietf.org/html/rfc7396#section-1[defined], that forces the removal of a particular key by setting its value to `null`. When user applied config contains `some_key: null` (or equivalent `some_key: ~`) it's interpreted as an instruction to remove `some_key`, often resulting in a malformed config. In Beat configs, this is often the case with link:https://www.elastic.co/guide/en/beats/filebeat/current/add-cloud-metadata.html[processors]. To avoid this:
+When `kubectl` is used to modify a resource, it calculates a diff between the user applied and the existing configuration. This diff has a special  link:https://tools.ietf.org/html/rfc7396#section-1[semantics], that forces the removal of a particular key by setting its value to `null`. When the user-applied configuration contains `some_key: null` (or equivalent `some_key: ~`), this is interpreted as an instruction to remove `some_key`, often resulting in a malformed configuration. In Beat configurations, this is often the case with link:https://www.elastic.co/guide/en/beats/filebeat/current/add-cloud-metadata.html[processors]. To avoid this:
 
-- change `some_key: null` to `some_key: {}` - if it doesn't change config semantics
-- instead of `config` use `configRef` and reference a Secret containing the config - Secret data doesn't go through the same diffing process
+- Change `some_key: null` to `some_key: {}`, if it doesn't change config semantics.
```

Comment:
```suggestion
- Use `some_key: {}` (empty map) or `some_key: []` (empty array) instead of `some_key: null` if doing so does not affect the behaviour. This might not be possible in all cases as some applications distinguish between null values and empty values and behave differently.
```